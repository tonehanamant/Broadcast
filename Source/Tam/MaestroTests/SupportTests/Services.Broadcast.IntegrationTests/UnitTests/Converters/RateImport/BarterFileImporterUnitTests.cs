using Common.Services.Repositories;
using Moq;
using NUnit.Framework;
using Services.Broadcast.Converters.RateImport;

namespace Services.Broadcast.IntegrationTests.UnitTests.Converters.RateImport
{
    [TestFixture]
    public class BarterFileImporterUnitTests
    {
        [Test]
        [TestCase(null, null)]
        [TestCase("", "")]
        [TestCase("WCBS-tv", "WCBS")]
        [TestCase("WCBS", "WCBS")]
        [TestCase("WCBS-TV", "WCBS")]
        [TestCase("WCBS-TV2", "WCBS")]
        [TestCase("WCBS-TV 123.4", "WCBS")]
        [TestCase("WCBS-DT", "WCBS")]
        [TestCase("WCBS-DT2", "WCBS")]
        [TestCase("WCBS-DT 123.4", "WCBS")]
        [TestCase("WCBS-LD", "WCBS")]
        [TestCase("WCBS-LD2", "WCBS")]
        [TestCase("WCBS-LD 123.4", "WCBS")]
        [TestCase("WCBS-CD", "WCBS")]
        [TestCase("WCBS-CD2", "WCBS")]
        [TestCase("WCBS-CD 123.4", "WCBS")]
        [TestCase("WCBS-LPTV", "WCBS")]
        [TestCase("WCBS-LPTV2", "WCBS")]
        [TestCase("WCBS-LPTV 123.4", "WCBS")]
        public void TransformStationCallsign(string rawCallsign, string expectedResult)
        {
            var dataRepositoryFactory = new Mock<IDataRepositoryFactory>();

            var fileImporter = new BarterFileImporter(
                dataRepositoryFactory.Object, null, null,
                null, null, null,
                null, null, null,
                null);

            var result = fileImporter._TransformStationCallsign(rawCallsign);

            if (expectedResult == null)
            {
                Assert.IsNull(result);
            }
            else
            {
                Assert.AreEqual(expectedResult, result);
            }
        }
    }
}