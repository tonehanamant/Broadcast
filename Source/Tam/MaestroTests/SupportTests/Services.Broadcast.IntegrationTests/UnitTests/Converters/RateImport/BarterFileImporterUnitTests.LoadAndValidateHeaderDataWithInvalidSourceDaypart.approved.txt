{
  "CreatedBy": null,
  "CreatedDate": "0001-01-01T00:00:00",
  "DataLines": [],
  "EffectiveDate": null,
  "EndDate": null,
  "FileName": "Barter_CNN_InvalidDaypart.xlsx",
  "FileStatus": 0,
  "Hash": null,
  "Header": {
    "Audience": {
      "_CategoryCode": 0,
      "_Code": null,
      "_Custom": false,
      "_Id": 31,
      "_Name": "Households",
      "_RangeEnd": null,
      "_RangeStart": null,
      "_SubCategoryCode": null,
      "IsDirty": true
    },
    "ContractedDaypartId": 1,
    "Cpm": 8.6,
    "DaypartCode": null,
    "EffectiveDate": "2018-01-01T00:00:00",
    "EndDate": "2018-03-20T00:00:00",
    "HutBookId": null,
    "NtiToNsiIncrease": null,
    "PlaybackType": 1,
    "ShareBookId": null,
    "SpotLengthId": null
  },
  "Id": 0,
  "InventoryGroups": [],
  "InventoryManifests": [],
  "InventorySource": {
    "Id": 1,
    "InventoryType": 5,
    "IsActive": true,
    "Name": "CNN"
  },
  "RowsProcessed": null,
  "UniqueIdentifier": null,
  "ValidationProblems": [
    "Not acceptable daypart code is specified",
    "Share book (2/1/808) is not in the correct format (MMM yy, MMM-yy, MMM/yy, yy-MMM, yy/MMM, MMM yyyy)",
    "Hut book (11/1/808) is not in the correct format (MMM yy, MMM-yy, MMM/yy, yy-MMM, yy/MMM, MMM yyyy)"
  ]
}